### PR TITLE
added functionalities for user management

### PR DESCRIPTION
**Pull request by AnikaAgiwal2711**
# Description
This PR includes adding of some functionalities related to user management like creating a new user, set it's role and removing a specific user.

# GitHub Issues
List the GitHub issues impacted by this PR:

| GitHub Issue # |
| -------------- |
| |

# Checklist:

- [X] I have performed a self-review of my own code to ensure there are no formatting, vetting, linting, or security issues
- [ ] I have verified that new and existing unit tests pass locally with my changes
- [X] I have not allowed coverage numbers to degenerate
- [ ] I have maintained at least 90% code coverage
- [X] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [X] I have added tests that prove my fix is effective or that my feature works
- [X] Backward compatibility is not broken

# How Has This Been Tested?
Since my changes includes adding of new functionalities without touching any existing functionality . So I created a test to check my functions and ran it successfully. See the ss below .
image

Also wrote the unit test as per the comment and ran it successfully. SS below
image

### CODE REVIEW

File: user.go [hunk 1]
```diff
@@ -36,3 +36,45 @@ func (s *System) GetUser() ([]types.User, error) {
 
 	return user, nil
 }
+
+// GetUserByID returns a specific user based on it's user id
+func (s *System) GetUserByID(userID string) (*types.User, error) {
```
Comment (khareRajshree) [inline user.go:41]: Is it possible to get user by name? If so, please add a method for the same or update the existing method to GetUserByIDName(id string, name string). You can refer to this, https://github.com/dell/goscaleio/blob/main/fs.go#L54

File: inttests/user_test.go [hunk 2]
```diff
@@ -13,9 +13,10 @@
 package inttests
 
 import (
-	"testing"
-
+	siotypes "github.com/dell/goscaleio/types/v1"
```
Comment (khareRajshree) [inline inttests/user_test.go:16]: Please add the unit tests as well.

**Review by khareRajshree — CHANGES_REQUESTED** [review]

File: user.go (hunk 1 above)
Comment (AnikaAgiwal2711) [reply user.go]: changed the exiting method to incorporate the name field also

File: inttests/user_test.go (hunk 2 above)
Comment (AnikaAgiwal2711) [reply inttests/user_test.go:16]: Done.

File: user_test.go [hunk 3]
```diff
@@ -0,0 +1,206 @@
...
+
+import (
+	"errors"
+	"math"
+	"net/http"
+	"net/http/httptest"
+	"testing"
+
+	types "github.com/dell/goscaleio/types/v1"
+	//"github.com/stretchr/testify/assert"
```
Comment (khareRajshree) [inline user_test.go:11]: please remove the commented code, if not needed.

**Review by khareRajshree — CHANGES_REQUESTED** [review]

File: user_test.go (hunk 3 above)
Comment (AnikaAgiwal2711) [reply user_test.go]: Done

**Review by khareRajshree — APPROVED** [review]

File: user.go [hunk 4]
```diff
@@ -36,3 +36,65 @@ func (s *System) GetUser() ([]types.User, error) {
...
+
+		return user, nil
+
+	} else {
+		allUsers, err := s.GetUser()
+		if err != nil {
+			return nil, err
+		}
+
+		for _, u := range allUsers {
```
Comment (VamsiSiddu-7) [inline user.go:64]: i think instead of u if it is user it will be nice

Comment (AnikaAgiwal2711) [reply user.go]: Done

**Review by VamsiSiddu-7 — APPROVED** [review]

**Pull request merged**